cirq/google/vis/heatmap.py

```diff
@@ -0,0 +1,250 @@
+        self.annot_map = annot_map
+        self.annot_kws = kws
+        return self
+
+    def set_annotation_format(self, annot_format: str, **kws: str) -> 'Heatmap':
+        """Sets a format string to format values for each qubit.
+
+        Args:
+            annot_format: the format string for formating values.
+            kws: keyword arguments to matplotlib.text.Text().
+        """
+        self.annot_map = {
+            key: format(value[1], annot_format)
+            for key, value in self.value_map.items()
+        }
+        self.annot_kws = kws
+        return self
+
+    def unset_annotation(self) -> 'Heatmap':
+        """Disables annotation. No texts are shown in cells."""
+        self.annot_map = {}
+        return self
+
+    def set_url_map(self, url_map: Mapping[QubitCoordinate, str]) -> 'Heatmap':
+        """Sets the URLs for each cell."""
+        self.url_map = url_map
+        return self
+
+    def unset_url_map(self) -> 'Heatmap':
+        """Disables URL. No URLs are associated with cells."""
+        self.url_map = {}
+        return self
+
+    def set_colorbar(self,
+                     position: str = 'right',
+                     size: str = '5%',
+                     pad: str = '2%',
+                     **kws: Any) -> 'Heatmap':
+        """Sets location and style of colorbar.
+
+        Args:
+            position: colorbar position, one of 'left'|'right'|'top'|'bottom'.
+            size: a string ending in '%' to specify the width of the colorbar.
+                Nominally, '100%' means the same width as the heatmap.
+            pad: a string ending in '%' to specify the space between the
+                colorbar and the heatmap.
+        """
+        self.plot_colorbar = True
+        self.colorbar_location_options = {
+            'position': position,
+            'size': size,
+            'pad': pad
+        }
+        self.colorbar_options = kws
+        return self
+
+    def unset_colorbar(self) -> 'Heatmap':
+        """Disables colorbar. No colorbar is drawn."""
+        self.plot_colorbar = False
+        return self
+
+    def set_colormap(self,
+                     colormap: Union[str, mpl.colors.Colormap] = 'viridis',
+                     vmin: Optional[float] = None,
+                     vmax: Optional[float] = None) -> 'Heatmap':
+        """Sets the colormap.
+
+        Args:
+            colormap: either a colormap name or a Colormap instance.
+            vmin: the minimum value to map to the minimum color. Default is
+                the minimum value in value_map.
+            vmax: the maximum value to map to the maximum color. Default is
+                the maximum value in value_map.
+        """
+        self.colormap = colormap
+        self.vmin = vmin
+        self.vmax = vmax
+        return self
+
+    def plot(self, ax: plt.Axes,
+             **kws: Any) -> Tuple[mpl_collections.Collection, pd.DataFrame]:
+        """Plots the heatmap on the given Axes.
+
```

Comment:
This would be a good place to document usage.

Reply:
A lot of usage are in the setters and unsetters. I've put a usage in the file level doc.